src/transformers/testing_utils.py

```diff
@@ -1570,6 +1572,34 @@ def LoggingLevel(level):
         transformers_logging.set_verbosity(orig_level)
 
 
+class TemporaryHubRepo:
+    """Create a temporary Hub repository and return its `RepoUrl` object. This is similar to
+    `tempfile.TemporaryDirectory` and can be used as a context manager.  For example:
+
+        with TemporaryHubRepo() as temp_repo:
+            ...
+
+    Upon exiting the context, the repository and everything contained in it are removed.
+
+    Example:
+
+    ```python
+    with TemporaryHubRepo() as temp_repo:
+        model.push_to_hub(tmp_repo.repo_id, token=self._token)
+    ```
+    """
+    def __init__(self, token=None):
```

Comment:
@Wauplin Here probably it's better to allow `create_repo_kwargs`?

Reply:
it could be added later if necessary but I don't think it is right? In `huggingface_hub` we have extra kwarg typically for `repo_type` but if `transformers` only gonna play with model repos, then it's not needed. Better to add when there is a need IMO